### PR TITLE
Adding task1 core components using different React ways

### PR DESCRIPTION
**Pull request by bandax**
React mentoring first task:

- Create react ecosystem.

- Create different component using React alternatives: Component, Functional, PureComponent, CreateElement

- Install developer tools to debug
dev tools

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+
+const App = () => (
+    <div>
+        <Header />        
+        <Footer />
+    </div>
+);
+
+export default App;
```
Comment (endiny) [inline src/App.jsx:13]: Please, add the empty line at the end of the file

File: src/Footer.jsx [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import React, { PureComponent } from 'react';
+
+export default class Footer extends PureComponent {
+    render(){
```
Comment (endiny) [inline src/Footer.jsx:4]: `() {`

File: src/Footer.jsx [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+        return <h4>Contact: administrator@movies.com</h4>
+    }
+}
```
Comment (endiny) [inline src/Footer.jsx:7]: empty line is missing

File: src/Footer.jsx [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import React, { PureComponent } from 'react';
+
+export default class Footer extends PureComponent {
```
Comment (endiny) [inline src/Footer.jsx:3]: I'd suggest to use named exports instead of default, so we could know the name during the import

File: src/Header.jsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+class Header extends React.Component{
```
Comment (endiny) [inline src/Header.jsx:3]: Let's keep classes definition in a single way. Either use `React.Component/PureComponent` or `Component/PureComponent` in the applicatino

File: src/Header.jsx [hunk 6]
```diff
@@ -0,0 +1,9 @@
...
+    render(){
```
Comment (endiny) [inline src/Header.jsx:4]: `() {`

File: src/index.js [hunk 7]
```diff
@@ -1,8 +1,23 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App.js';
+import App from './App.jsx';
+
+const ShowDate = () =>{
+    return (
+        <p>Today is {new Date().toLocaleString()}</p>
+    )
```
Comment (endiny) [inline src/index.js:8]: missing semicolon

File: src/Header.jsx [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+        return <h1>Welcome to Movie Catalog</h1>
```
Comment (endiny) [inline src/Header.jsx:5]: missing semicolon

File: src/index.js [hunk 9]
```diff
@@ -1,8 +1,23 @@
...
 
 ReactDOM.render(
-    <App />,
+    <div>
+        <ShowDate />
+        <ShowMessage />
+        <App />
+    </div>,
     document.getElementById('root')
 );
```
Comment (endiny) [inline src/index.js:23]: empty line is missing

File: src/index.js [hunk 10]
```diff
@@ -1,8 +1,23 @@
...
+const ShowDate = () =>{
+    return (
+        <p>Today is {new Date().toLocaleString()}</p>
+    )
+};
+
+const ShowMessage = () => {
+    return (React.createElement('div', null, `Hello World`));
+};
+
```
Comment (endiny) [inline src/index.js:14]: * Please extract that components to another file
* Components names should not contain verbs in it. I believe `Date` and `Message` (or something else) are way better

File: src/index.js [hunk 11]
```diff
@@ -1,8 +1,23 @@
...
+import App from './App.jsx';
+
+const ShowDate = () =>{
+    return (
+        <p>Today is {new Date().toLocaleString()}</p>
+    )
+};
+
+const ShowMessage = () => {
+    return (React.createElement('div', null, `Hello World`));
```
Comment (endiny) [inline src/index.js:12]: no parentheses are needed

File: src/index.js [hunk 12]
```diff
@@ -1,8 +1,23 @@
...
+};
+
 
 ReactDOM.render(
-    <App />,
+    <div>
+        <ShowDate />
+        <ShowMessage />
+        <App />
+    </div>,
```
Comment (endiny) [inline src/index.js:21]: I think that content can be moved to the `<App />` component

File: src/App.jsx (hunk 1 above)
Comment (bandax) [reply src/App.jsx]: done.

File: src/Footer.jsx (hunk 2 above)
Comment (bandax) [reply src/Footer.jsx]: done.

File: src/Footer.jsx (hunk 3 above)
Comment (bandax) [reply src/Footer.jsx]: done.

File: src/Footer.jsx (hunk 4 above)
Comment (bandax) [reply src/Footer.jsx]: I applied the export named style for all components

File: src/Header.jsx (hunk 5 above)
Comment (bandax) [reply src/Header.jsx]: done.

File: src/Header.jsx (hunk 8 above)
Comment (bandax) [reply src/Header.jsx]: added.

File: src/index.js (hunk 9 above)
Comment (bandax) [reply src/index.js]: done.

File: src/index.js (hunk 10 above)
Comment (bandax) [reply src/index.js]: Created two new components and integrated into different components

File: src/index.js (hunk 12 above)
Comment (bandax) [reply src/index.js]: done.

File: src/Footer.jsx [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import React, { PureComponent } from 'react';
+
+class Footer extends PureComponent {
```
Comment (endiny) [inline src/Footer.jsx:3]: can be shorted to `export class Footer...` so the last line won't be needed

File: src/index.js [hunk 14]
```diff
@@ -1,8 +1,8 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
-import App from './App.js';
+import { App } from './App.jsx';
 
-ReactDOM.render(
+ReactDOM.render(
```
Comment (endiny) [inline src/index.js:5]: Please remove extra spaces/tabs

File: src/Footer.jsx (hunk 13 above)
Comment (endiny) [reply src/Footer.jsx:3]: same for other exports

File: src/index.js (hunk 14 above)
Comment (bandax) [reply src/index.js]: done.

File: src/Footer.jsx (hunk 13 above)
Comment (bandax) [reply src/Footer.jsx]: done.

**Pull request merged**